src/lib/x509/certstor.cpp

```diff
@@ -213,4 +214,135 @@ Certificate_Store_In_Memory::Certificate_Store_In_Memory(const std::string& dir)
    }
 #endif
 
+/**
+ * Internal class implementation (i.e. Pimpl) to keep the required platform-
+ * dependent members of Flatfile_Certificate_Store contained in this compilation
+ * unit.
+ */
+class Flatfile_Certificate_Store_Impl
+   {
+   public:
+      Flatfile_Certificate_Store_Impl(const std::string& system_store) :
+         m_system_store(system_store)
+         {
+
+         if (m_system_store.empty()) { return; }
+
+         DataSource_Stream file(m_system_store);
+         for(const secure_vector<uint8_t> der : PEM_Code::decode_all(file))
+            {
+            m_certificates.push_back(std::make_shared<X509_Certificate>(der.data(), der.size()));
+            }
+         }
+
+   public:
+      const std::string m_system_store;
+      std::vector<std::shared_ptr<const X509_Certificate>> m_certificates;
+   };
+
+Flatfile_Certificate_Store::Flatfile_Certificate_Store() :
+   m_impl(std::make_shared<Flatfile_Certificate_Store_Impl>(""))
+   {
+   }
+
+#if defined(BOTAN_TARGET_OS_HAS_FILESYSTEM)
+Flatfile_Certificate_Store::Flatfile_Certificate_Store(const std::string& file) :
+   m_impl(std::make_shared<Flatfile_Certificate_Store_Impl>(file))
+   {
+   }
+#endif
```

Comment:
The whole `Flatfile_Certificate_Store` won't be very useful if no file system is available on the target platform. Maybe disable the whole class in this case?